Add number of guests

jehousoh opened the pull request:
- Prefix: ng/
- Adds the field "numberOfGuests" to the Guest class
- Enables the ability to edit the numberOfGuests
- Create a label to display "No. of Guests:" on each Guest
- Add test cases to test numberOfGuests field

closes #40

jehousoh committed:
Add NumberOfGuests field

Let's add a NumberOfGuests class and integrate it into the other
classes.

jehousoh committed:
Update test cases

Test cases did not work due to the added field "NumberOfGuests".

Let's fix the old tests cases for it to be able to test whether the
classes are working again.

jehousoh committed:
Fix checkstyle

There is some errors regarding checkstyle.

Let's fix the code to fit into the checkstyle format.

jehousoh committed:
Add NumberOfGuests test cases

There is no test cases for the NumberOfGuests field.

Let's add test cases to test if the field works completely.

jehousoh committed:
Merge branch 'master' into add-NumberOfGuests

jehousoh committed:
Merge branch 'master' into add-NumberOfGuests

jehousoh committed:
Fix checkstyle issue and merge conflicts

There were some problems with the import statements and merge conflicts
in the PersonUtil class.

Let's
fix all the checkstyle issues that are present
resolve the merge conflict in the PersonUtil class

bryanhce commented on src/main/java/seedu/guest/model/guest/NumberOfGuests.java:
```diff
@@ -0,0 +1,53 @@
+import static seedu.guest.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the total number of guests a Guest has in the address book.
```
Should this be guest book instead of address book?

jehousoh replied:
Yes it should. Fixed on the latest commit.

bryanhce reviewed (changes requested):
Nice work! Just a minor nit to fix.

kangzongxian commented on src/main/java/seedu/guest/model/guest/NumberOfGuests.java:
```diff
@@ -0,0 +1,53 @@
+import static seedu.guest.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the total number of guests a Guest has in the address book.
```
Should this be guest book instead of address book?

jehousoh replied:
Yes it should. Fixed on the latest commit.

kangzongxian commented on src/test/java/seedu/guest/logic/commands/AddCommandIntegrationTest.java:
```diff
@@ -27,7 +27,7 @@ public void setUp() {
 
     @Test
     public void execute_newPerson_success() {
```
Should this method name be newGuest instead of newPerson?

kangzongxian replied:
Maybe it is irrelevant for now for your PR as it seems you were refactoring to add the number of guests as a field!

jehousoh replied:
Will be done by Bryan under refactoring tests.

jehousoh committed:
Fix minor typo

kangzongxian reviewed (approved):
Looks good to me!

JustinPeng13 commented on src/main/java/seedu/guest/logic/commands/EditCommand.java:
```diff
@@ -40,6 +42,7 @@ public class EditCommand extends Command {
             + "[" + PREFIX_NAME + "NAME] "
             + "[" + PREFIX_PHONE + "PHONE] "
             + "[" + PREFIX_EMAIL + "EMAIL] "
+            + "[" + PREFIX_NUMBER_OF_GUESTS + "NUMBER OF GUESTS] "
```
Should this be NUMBER_OF_GUESTS?

jehousoh replied:
Yes it should. Changed made in the recent commit.

jehousoh committed:
Change NUMBER OF GUESTS to NUMBER_OF_GUESTS

JustinPeng13 commented on src/test/java/seedu/guest/logic/parser/AddCommandParserTest.java:
```diff
@@ -36,101 +40,128 @@
         // all prefixes missing
-        assertParseFailure(parser, VALID_NAME_BOB + VALID_PHONE_BOB + VALID_EMAIL_BOB + VALID_ADDRESS_BOB,
+        assertParseFailure(parser, VALID_NAME_BOB + VALID_PHONE_BOB + VALID_EMAIL_BOB
+                + NUMBER_OF_GUESTS_DESC_BOB + VALID_ADDRESS_BOB,
```
Should this be VALID_NUMBER_OF_GUESTS_BOB?

jehousoh replied:
Yes it should. Changed made in the recent commit.

JustinPeng13 commented on src/test/java/seedu/guest/logic/parser/AddCommandParserTest.java:
```diff
@@ -36,101 +40,128 @@
+                + NUMBER_OF_GUESTS_DESC_BOB + INVALID_ADDRESS_DESC,
                 Name.MESSAGE_CONSTRAINTS);
 
         // non-empty preamble
```
Should we add NUMBER_OF_GUESTS_DESC_BOB to this test case as well?

jehousoh replied:
Yes we should. Changed made in the recent commit.

JustinPeng13 commented on src/test/java/seedu/guest/logic/parser/EditCommandParserTest.java:
```diff
@@ -169,12 +183,12 @@ public void parse_oneFieldSpecified_success() {
         String userInput = targetIndex.getOneBased() + PHONE_DESC_AMY + ADDRESS_DESC_AMY + EMAIL_DESC_AMY
-                + TAG_DESC_FRIEND + PHONE_DESC_AMY + ADDRESS_DESC_AMY + EMAIL_DESC_AMY + TAG_DESC_FRIEND
-                + PHONE_DESC_BOB + ADDRESS_DESC_BOB + EMAIL_DESC_BOB + TAG_DESC_HUSBAND;
+                + NUMBER_OF_GUESTS_DESC_AMY + PHONE_DESC_AMY + ADDRESS_DESC_AMY + EMAIL_DESC_AMY + TAG_DESC_FRIEND
```
Should there be 2 NUMBER_OF_GUESTS_DESC_AMY and 1 NUMBER_OF_GUESTS_DESC_BOB? Also keep the 2 TAG_DESC_FRIEND?

jehousoh replied:
Yes there should. Changed made in the recent commit.

JustinPeng13 commented on src/test/java/seedu/guest/model/guest/NumberOfGuestsTest.java:
```diff
@@ -0,0 +1,40 @@
+
+    @Test
+    public void constructor_invalidNumberOfGuests_throwsIllegalArgumentException() {
+        String invalidNumberOfGuests = "-1";
```
Should this be "" in line with other tests? I think the actual invalid numbers are the ones in isValidNumberOfGuests test

jehousoh replied:
Yes it should. Changed made in the recent commit.

JustinPeng13 commented on src/test/java/seedu/guest/storage/JsonAdaptedGuestTest.java:
```diff
@@ -27,6 +27,7 @@ public class JsonAdaptedGuestTest {
     private static final String VALID_NAME = BENSON.getName().toString();
```
Should we include INVALID_NUMBER_OF_GUESTS and tests for invalidNumberOfGuests and nullNumberOfGuests?

jehousoh replied:
Yes we should. Changed made in the recent commit.

jehousoh committed:
Fix tests that were missed

Some tests did not have NumberOfGuests tested

Let's add NumberOfGuests to those test cases
